uninstall.php

```diff
@@ -75,28 +75,15 @@
 	return;
 }
 
-require_once WEBSTORIES_PLUGIN_DIR_PATH . '/includes/uninstall.php';
-
-if ( is_multisite() ) {
-	\Google\Web_Stories\delete_site_options();
-	$site_ids = get_sites(
-		[
-			'fields'                 => 'ids',
-			'number'                 => '',
-			'update_site_cache'      => false,
-			'update_site_meta_cache' => false,
-		]
-	);
-
-	foreach ( $site_ids as $site_id ) {
-		// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.switch_to_blog_switch_to_blog
-		switch_to_blog( $site_id );
-
-		\Google\Web_Stories\delete_site();
-	}
-
-	restore_current_blog();
-} else {
-	\Google\Web_Stories\delete_site();
+/**
+ * Defer running uninstall until every service is registered.
+ *
+ * @since 1.26.0
+ */
+function web_stories_uninstall(): void {
+	$ws_plugin = \Google\Web_Stories\PluginFactory::create();
+	$ws_plugin->register();
+	$ws_plugin->on_site_uninstall();
 }
-\Google\Web_Stories\delete_stories_user_meta();
+add_action( 'shutdown', 'web_stories_uninstall' );
```

Comment:
What's the purpose of this? Why only do the uninstall on shutdown?

Reply:
> Defer running uninstall until every service is registered.

See comment above. Sadly I could not find a clean way to only register services without delays or register methods. The cleanest way was to just defer to the latest hook and run the uninstall methods there.

Reply:
Deferring until shutdown seems like a hacky workaround to me though.

I tend to think that it must be possible to adjust the `Uninstaller` so that all services are registered straight away, ignoring `Delayed` and such.

Reply:
Uninstaller was removed as I could not get it to work.

Reply:
It is a hacky workaround for our setup. Uninstall is an uncommon use case. As it is rarely used, this workaround to bootstrap works surprisingly well and is only for this one place.

Reply:
> Uninstaller was removed as I could not get it to work.

Hmm I see. Would have been nice. Too bad.